src/shogun/base/init.cpp

```diff
@@ -38,6 +37,7 @@ namespace shogun
 	SGIO* sg_io=NULL;
 	Version* sg_version=NULL;
 	CMath* sg_math=NULL;
+	__int32_t sg_random_seed = shogun::CRandom::generate_seed();
```

Comment:
mmm, the logic in here is a bit of odd in here. I define the global random seed in init.cpp because here is where we define global variable. Then I do setter and getter function in SGObject in [here](https://github.com/shogun-toolbox/shogun/pull/3906/files#diff-1689b0bd3ceb28355cc3c05ed7ffe4f0R504) due to we need to reference shogun/base/init.h every time if we want to call `set_global_seed()` or getter otherwise.